core/domain/story_domain.py

```diff
@@ -666,7 +666,10 @@ def get_ordered_nodes(self) -> List[StoryNode]:
         """
         # Ruling out the possibility of None for mypy type checking.
         assert self.initial_node_id is not None
-        initial_index = self.get_node_index(self.initial_node_id)
+        initial_index = self.get_node_index(self.initial_node_id, strict=False)
+        # TODO(#15639): Not sure if this condition is a correct solution.
```

Comment:
This seems wrong to me. Why is the initial node ID not having an index? Is the data on the test server broken in some way? If so we should fix that there, rather than here.